### PR TITLE
Ospray 1.8

### PR DESCRIPTION
**Pull request by hernando**

### CODE REVIEW

File: engines/ospray/ispc/geometry/Cylinders.ispc [hunk 1]
```diff
@@ -64,9 +63,14 @@ unmasked void Cylinders_bounds(const RTCBoundsFunctionArguments* uniform args)
              : self->radius;
     uniform vec3f v0 = *((vec3f * uniform)(cylinderPtr + self->offset_v0));
     uniform vec3f v1 = *((vec3f * uniform)(cylinderPtr + self->offset_v1));
-    // TODO less conservative bounds
+
+    uniform vec3f a2 = sqr(v0 - v1);
```
Comment (karjonas) [inline engines/ospray/ispc/geometry/Cylinders.ispc]: I cannot figure out what you are calculating here, can you name these variables to something meaningful?

File: engines/ospray/ispc/geometry/Cylinders.ispc [hunk 2]
```diff
@@ -84,73 +88,70 @@ void Cylinders_intersect_kernel(const RTCIntersectFunctionNArguments* uniform
...
         ray->t = t_out;
     }
 
     if (hit)
     {
-        if (isOcclusionTest)
-        {
-            ray->t = neg_inf;
-        }
+        cif(isOcclusionTest) { ray->t = neg_inf; }
```
Comment (karjonas) [inline engines/ospray/ispc/geometry/Cylinders.ispc]: cif?

File: engines/ospray/ispc/geometry/Cylinders.ispc [hunk 3]
```diff
@@ -84,73 +88,70 @@ void Cylinders_intersect_kernel(const RTCIntersectFunctionNArguments* uniform
...
+    // rsd (and thus tA and tB) can be NaN if the ray is perfectly perpendicular
+    // to the cylinder; filter NaNs by having ray t as 2nd argument to min/max
+    const float tmin = max(min(tA, tB), ray->t0);
+    const float tmax = min(max(tA, tB), ray->t);
 
     bool hit = false;
-
-    if (t_in >= (tAB0) && t_in <= (tAB1))
+    // avoid short-circuiting &&
+    if (and(tmin < t_in, t_in <= tmax))
```
Comment (karjonas) [inline engines/ospray/ispc/geometry/Cylinders.ispc]: Why this syntax?

Comment (hernando) [reply engines/ospray/ispc/geometry/Cylinders.ispc]: To avoid short-circuiting && :)

**Comment by hernando** [comment]
It's all copy-pasta plus the carbonara, because there's a [bug](https://github.com/ospray/ospray/issues/302) in their code.
The cylinder code was already taken from OSPRay, I have just updated it to their latest mater (+ the fix). I won't create a separate commit for it as it's an indivisible part of the adaptations to 1.8. Otherwise tests don't pass.

File: engines/ospray/ispc/geometry/Cylinders.ispc (hunk 1 above)
Comment (hernando) [reply engines/ospray/ispc/geometry/Cylinders.ispc]: I with I could understand this myself.

File: engines/ospray/CMakeLists.txt [hunk 4]
```diff
@@ -91,15 +91,17 @@ set_source_files_properties(
     PROPERTIES COMPILE_FLAGS -Wno-deprecated-declarations)
 
 set(BRAYNSOSPRAYENGINE_LINK_LIBRARIES
-    PRIVATE braynsEngine braynsParameters ${OSPRAY_LIBRARIES}
+    PRIVATE braynsEngine braynsParameters ospray::ospray_common ospray::ospray
+    # Public linkage needed for tranfer function test
```
Comment (tribal-tec) [inline engines/ospray/CMakeLists.txt]: typo `tranfer`

**Comment by karjonas** [comment]
retest this please

**Comment by ppodhajski** [comment]
retest this please

**Comment by ppodhajski** [comment]
retest this please

**Comment by ppodhajski** [comment]
retest this please

**Comment by ppodhajski** [comment]
retest this please

**Comment by ppodhajski** [comment]
retest this please

**Comment by ppodhajski** [comment]
retest this please

**Comment by ppodhajski** [comment]
retest this please

**Comment by ppodhajski** [comment]
retest this please

**Comment by ppodhajski** [comment]
retest this please

**Comment by ppodhajski** [comment]
retest this please

**Comment by ppodhajski** [comment]
retest this please

**Comment by ppodhajski** [comment]
retest this please

**Comment by ppodhajski** [comment]
retest this please

**Comment by ppodhajski** [comment]
retest this please

**Comment by ppodhajski** [comment]
retest this please

**Comment by ppodhajski** [comment]
retest this please

**Comment by ppodhajski** [comment]
retest this please

**Pull request merged**